_posts/2016-09-14-programming-model.md

```diff
@@ -0,0 +1,128 @@
+---
+layout: post
+title: Programming Model
+---
+
+
+## Overview
+
+Graphics APIs, much like programming languages, come with their own model of thinking. Understanding this model allows to use the API efficiently, but also to see the rationale behind the core design decisions that would otherwise be non-obvious. GFX has historically been confusing for newcomers, especially those coming from OpenGL background. We've come a long way too - first GFX was very similar to GL, but our thinking went through several revolutions since then. We are not done yet, but given that our graphics backend support has quadrupled in 2016 ([Dx11](https://github.com/gfx-rs/gfx/tree/master/src/backend/dx11), [Metal](https://github.com/gfx-rs/gfx/tree/master/src/backend/metal), [Vulkan](https://github.com/gfx-rs/gfx/tree/master/src/backend/vulkan)), it would be considerably more difficult to make radical changes now.
```

Comment:
Does it need a short description of what gfx is?

> Understanding this model allows to use the API efficiently

Allows who to use it? Talking about GFX specifically or all kinds of graphics APIs?

> especially those coming from **an** OpenGL background

> We've come a long way too - **at** first GFX was very similar to 

> but our thinking went through several revolutions since then.

s/revolutions/iterations/?

> but given that our graphics backend support has quadrupled in 2016

number of backends we support has quadrupled?